### ISSUE TITLE
Unable to remove /home/USER/.steam/CONFIG/SteamAppData.vdf!

### ISSUE THREAD

**Issue by xksteven** [ValveSoftware/steam-for-linux#4559]
#### Your system information
- Steam client version:
- Distribution (e.g. Ubuntu): Ubuntu 16.04
- Opted into Steam client beta?: Yes
- Have you checked for system updates?:  Yes
#### Please describe your issue in as much detail as possible:

Steam wouldn't launch
#### Steps for reproducing this issue:
1. Start steam from command-line
   Unable to remove /home/USER/.steam/CONFIG/SteamAppData.vdf!

**Comment by slackerbob** [comment]
I'm having the same problem. symlinking /home/USER/.steam/CONFIG to /home/USER/.steam/config does not fix the issue, as there is no SteamAppData.vdf in config (or anywhere that I could find).

**Comment by xksteven** [comment]
I was able to "fix"  it by moving .steam folder to .steam.bkup
running in the terminal 

    steam

Then steam will redownload everything.  from there I moved the .steam.bkup/steam into the new .steam folder (it's the folder that contained all of my game data).  I also needed to remove libstdc++.so from  ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/ and  ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/

Not sure if it'll break again though.

**Comment by Theswweet** [comment]
I'm having this issue as well.

**Comment by LexBarringer** [comment]
The folder and file in question isn't really the issue here, I use Ubuntu 16.04 (Xubuntu desktop). First off, the reason why it can't delete it as you stated the file doesn't exist in that folder, which is actually nothing more than a soft error; a warning.

The CONFIG folder doesn't exist; this is because, "CONFIG" and "config" and "Config" on Microsoft Windows are all the same folder but on the *NIX and Linux they're completely separate folders. This is just an oversight and remnants from a Microsoft Windows build that should have been omitted for the Linux and Apple OS X. 

Now, I will speak to the problem as to why your system may not bootstrap Steam due to the stdc++.so.6 files from within the Steam path. The symbolic links that are in the directory are wrong for anything beyond Ubuntu 12.04 (or forks) and especially for the x86_64 edition and beyond.

This is what I do anytime I have to do a new install on the computers with Steam.

cd $HOME/.steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu
rm libstdc++.so.6
cd $HOME/.steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu
rm libstdc++.so.6

This is actually a very well known problem in the Ubuntu and forks communities as is the fix. All the above does is get to the directories in question and delete (remove) the symlink file and then Steam will use the system libraries under the same name. 

Your problem with the beta in and of itself isn't indicative of having trouble with said libstdc++.so.6 symlinks in Debian Testing, that's another separate issue. There in lies the rub, you're using a Debian Testing or the Unstable branch with a beta client from Valve Software. I know you mean well but you can't actually tell if it's your system, the Steam beta or a combination of the two that is the actual cause of your ordeal.

I use the beta Steam client myself but am using Xubuntu 16.04 LTS x86_64 (64-bit) and no other beta packages nor backports are installed, I prefer a rock solid and stable computer to test this software. 
 
To better help Valve zero in on the problem you're having look for "hardinfo" in your download repositories. 

sudo apt-get update && sudo apt-cache search hardinfo

If you find that in your repositories, download it, it's quite handy for troubleshooting teams that include programmers. You can save what it finds in an .htm file which can be view from a web browser.

Note: If you find goofy errors in Valve Software's Steam client, that's what this is all about, that's why Valve created this software and having us post our errors, issues and possible pull requests if we have a fix in regards to say a Python script they're using.

**Comment by LexBarringer** [comment]
(For the TL;DR crowd

Basically, what I am saying is;

Don't worry about this message in console for Apple OS X and Linux (including SteamOS).

"Unable to remove /home/user/.steam/CONFIG/SteamAppData.vdf!"

It's just a warning, not an actual error, it has no bearing on the performance on your installation of Steam, it does nothing to your Steam client.

**Comment by ErikViWe** [comment]
Well, i got the same Problem while installing Steam on Ubuntu 16.10.
What i did is, i created a new folder called "CONFIG" and copied all files from "config" to "CONFIG".
I recognized that Steam writes some files into "config" and others into "CONFIG". This causes several problems.

The result is that i can't login to my Steam account anymore and this makes the problem to a serious issue, not to a warning.

If anyone has a solution to this Problem, please let me know.

**Comment by LexBarringer** [comment]
So, let's get this one straight. You created the CONFIG folder and copied all the contents into the new CONFIG folder from the config directory?

As in:
cp ~/.steam/config/  ~/.steam/CONFIG

In terminal type this:

rm -rf ~/.steam/CONFIG

What that will do is delete the CONFIG directory (in all upper caps).

Then attempt to run Steam in terminal, by typing the name:

steam

Then when it hangs do cut and paste the errors you see on the screen. As I said before the CONFIG directory is for a Microsoft Windows version of the Steam Client, it was incorrectly specified during a build, it doesn't have anything to do with the Linux or OS X build of the same software.

Basically, you have two problems, we're getting rid of one of them. When I see the result of the other, we'll get a better understanding on how Valve Software or I can help you.

**Comment by ErikViWe** [comment]
Thanks for the quick and detailed response.

To summarize: i removed the CONFIG folder and all files, in the way you told me.
Then i started Steam with the command "steam" in the terminal.
First output on the terminal was:

Running Steam on ubuntu 16.10 64-bit
STEAM_RUNTIME is enabled automatically
[2016-11-09 08:57:53] Startup - updater built Oct 13 2016 00:47:16
[2016-11-09 08:57:53] Verifying installation...
[2016-11-09 08:57:53] Verification complete

Then the classic login screen poped up and i tried to log in with my account name an password.
The next Terminal output was:

Unable to remove /home/erik/.steam/CONFIG/SteamAppData.vdf!

A new window with the title "Connection Error" appeared and asked me to retry, start the offline mode or to quit Steam.
At this point i quit Steam and don't know what to do.
I tried to retry but the result stays the same. Starting in offline mode is not what i want. To quit the program is the only option.

The ~/.steam/error.log shows following information:

rm: das Entfernen von '/home/erik/.steam/steam' ist nicht möglich: Ist ein Verzeichnis
rm: das Entfernen von '/home/erik/.steam/bin' ist nicht möglich: Ist ein Verzeichnis
/bin/bash: /home/erik/.steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"
Gtk-Message: Failed to load module "unity-gtk-module"
Installing breakpad exception handler for appid(steam)/version(1476379980)
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 80: saw unknown, expected number
[1109/085753:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak
[1109/085753:ERROR:browser_main_loop.cc(217)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476319626)
[1109/085753:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Created shared memory when not owner SteamController_Shared_mem
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Generating new string page texture 7: 128x256, total string texture memory is 131,07 KB
Generating new string page texture 8: 48x256, total string texture memory is 180,22 KB
Generating new string page texture 9: 256x256, total string texture memory is 442,37 KB
Generating new string page texture 10: 64x256, total string texture memory is 507,90 KB
Generating new string page texture 11: 8x256, total string texture memory is 516,10 KB
Generating new string page texture 12: 16x256, total string texture memory is 532,48 KB
Generating new string page texture 13: 24x256, total string texture memory is 557,06 KB
Generating new string page texture 14: 32x256, total string texture memory is 589,82 KB
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Generating new string page texture 16: 384x256, total string texture memory is 983,04 KB
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)

**Comment by LexBarringer** [comment]
Okay, I'll break each section down for you.

**Normal:**

Running Steam on ubuntu 16.10 64-bit
STEAM_RUNTIME is enabled automatically
[2016-11-09 08:57:53] Startup - updater built Oct 13 2016 00:47:16
[2016-11-09 08:57:53] Verifying installation...
[2016-11-09 08:57:53] Verification complete

Then the classic login screen poped up and i tried to log in with my account name an password.
The next Terminal output was:

**Don't worry, this is a build error and doesn't affect you:**

Unable to remove /home/erik/.steam/CONFIG/SteamAppData.vdf!

A new window with the title "Connection Error" appeared and asked me to retry, start the offline mode or to quit Steam.
At this point i quit Steam and don't know what to do.
I tried to retry but the result stays the same. Starting in offline mode is not what i want. To quit the program is the only option.

The ~/.steam/error.log shows following information:

**Error in Valve Software's scripting logic, not a concern as of right now:**

rm: das Entfernen von '/home/erik/.steam/steam' ist nicht möglich: Ist ein Verzeichnis
rm: das Entfernen von '/home/erik/.steam/bin' ist nicht möglich: Ist ein Verzeichnis
/bin/bash: /home/erik/.steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)

**More than likely your problem is right here: (These are Perl runtime related files)**

grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc

**These are irritating warnings but not critical:**

Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"
Gtk-Message: Failed to load module "unity-gtk-module"
Installing breakpad exception handler for appid(steam)/version(1476379980)
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 80: saw unknown, expected number

**Bugs in relation to the embedded browser that is used inside Steam, not critical for operation**

[1109/085753:ERROR:browser_main_loop.cc(217)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
[1109/085753:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak

**Normal stuff:**

Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476319626)
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Created shared memory when not owner SteamController_Shared_mem
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Generating new string page texture 7: 128x256, total string texture memory is 131,07 KB
Generating new string page texture 8: 48x256, total string texture memory is 180,22 KB
Generating new string page texture 9: 256x256, total string texture memory is 442,37 KB
Generating new string page texture 10: 64x256, total string texture memory is 507,90 KB
Generating new string page texture 11: 8x256, total string texture memory is 516,10 KB
Generating new string page texture 12: 16x256, total string texture memory is 532,48 KB
Generating new string page texture 13: 24x256, total string texture memory is 557,06 KB
Generating new string page texture 14: 32x256, total string texture memory is 589,82 KB
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Generating new string page texture 16: 384x256, total string texture memory is 983,04 KB
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)


So... What to do about that one area I noticed that is probably causing your connection errors and other annoyances?

Do the following in terminal:

sudo apt update
sudo apt install libpcre3 libpcre32-3 libpcrecpp0v5 libatk-adaptor libgail-common unity-gtk-module-*
sudo apt dist-upgrade

If you're curous why I'm not using apt-get, it's just more typing and I get the same result with apt and a progress bar at the bottom, which is a plus. 

What this is doing is installing the pcre run-time libraries for Perl 5 that seem to be missing from your installation, which by default is a part of the installation. Perhaps you had another application remove these packages by accident and thus causing this problem. These libraries have to do with Perl scripting in regards to network communications. If they're not in place and working, that could explain why Steam can't pass your login information to their servers because the files are missing. 

The last one is dist-upgrade, this means distribution upgrade, it bumps you up to the next point release of said distribution. For example; Ubuntu 16.04 when you dist-upgrade then becomes 16.04.1. With Yakkity Yak, there is no point release yet but using the dist-upgrade will give you the newest stable files, more so than the "apt upgrade" will.

Once you're done with this, start Steam again in terminal:

steam

Then let it do it's thing, see if you can connect and authenticate to your online account. If you can, close Steam normally, then start it via your desktop icon or in your menus and continue.

**Optional:**

I also would like to know if you're using the open source drivers that came with Yakkity Yak or if you're using a restricted driver from Nvidia, AMD or Intel. Depending on your answer I may have something else to help you increase your speed on your computer for rendering if you're using the open source that came with it.

**Comment by ErikViWe** [comment]
Hey, I installed the mentioned packages with:

sudo apt update
sudo apt install libpcre3 libpcre32-3 libpcrecpp0v5 libatk-adaptor libgail-common unity-gtk-module-*
sudo apt dist-upgrade

But Steam does still not work. The terminal output is the same, but as you said, that is not a problem.
The error.log is nearly the same, the three error where you thought the problem would be are still there:

grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc

(I will put the full error.log to the bottom of this post)

I do not have any clue, where to search next.

**Driver:**
I am using the nouveau driver. 

**error.log:**
rm: das Entfernen von '/home/erik/.steam/steam' ist nicht möglich: Ist ein Verzeichnis
rm: das Entfernen von '/home/erik/.steam/bin' ist nicht möglich: Ist ein Verzeichnis
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
/bin/bash: /home/erik/.steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)
grep: symbol lookup error: grep: undefined symbol: pcre_jit_stack_alloc
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"
Gtk-Message: Failed to load module "unity-gtk-module"
Installing breakpad exception handler for appid(steam)/version(1476379980)
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 80: saw unknown, expected number
[1110/082134:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak
[1110/082134:ERROR:browser_main_loop.cc(217)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476319626)
[1110/082134:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Created shared memory when not owner SteamController_Shared_mem
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Generating new string page texture 7: 128x256, total string texture memory is 131,07 KB
Generating new string page texture 8: 48x256, total string texture memory is 180,22 KB
Generating new string page texture 9: 256x256, total string texture memory is 442,37 KB
Generating new string page texture 10: 64x256, total string texture memory is 507,90 KB
Generating new string page texture 11: 8x256, total string texture memory is 516,10 KB
Generating new string page texture 12: 16x256, total string texture memory is 532,48 KB
Generating new string page texture 13: 24x256, total string texture memory is 557,06 KB
Generating new string page texture 14: 32x256, total string texture memory is 589,82 KB
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Installing breakpad exception handler for appid(steam)/version(1476379980)
Generating new string page texture 16: 384x256, total string texture memory is 983,04 KB
[1110/082150:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak
[1110/082150:ERROR:browser_main_loop.cc(217)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476319626)
[1110/082150:ERROR:main_delegate.cc(779)] Could not load cef_extensions.pak
Installing breakpad exception handler for appid(steamwebhelper)/version(20161013004706)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)
Installing breakpad exception handler for appid(steamwebhelper)/version(1476379980)

**Comment by LexBarringer** [comment]
There are another 3 glitches that can cause that problem. Try this in terminal.

LD_PRELOAD='/usr/$LIB/libstdc++.so.6' steam

If this starts Steam and you're able to login, your computer can't tell which version of the file to load, one in Steam's own path or the system path. What this is doing is loading it in your system's path which is the way it should be regardless.

**Comment by ErikViWe** [comment]
Yes, it works! Thank you very much. I'll let them know about this.

**Comment by LexBarringer** [comment]
Okay, that's good to hear I got your Steam installation working again. 

This problem occurs when the system libraries are different than the local libraries are in Steam. By default, Valve Software will use their own older versions of the libraries. As a rule of thumb, this is frowned upon in the open source market, breeds incompatibility with newer system and it's harder to support, if something should go wrong.

On another note, were you able to find the Steam shortcut and drag it to the desktop to edit the launcher? I assume by your answer that is correct but I don't want to jump the gun.

**Comment by LexBarringer** [comment]
@xksteven Are you still having problem getting Steam to work?

If you would please, open a terminal and run Steam directly, then cut and paste the output you get from Steam.

Type 'steam', without the quotes in terminal and hit Enter.

**Comment by xksteven** [comment]
I was able to reinstall steam and I deleted the local steam runtime c libraries.  If no one else is having the issue I can close this thread.

**Comment by LexBarringer** [comment]
Go ahead and close it out, Steven. If there is a new issue, people can open a new one and reference this issue if it applies.
